src/ceph-volume/ceph_volume/tests/util/test_system.py

```diff
@@ -216,3 +216,56 @@ def test_warnings_when_executable_isnt_matched(self, monkeypatch, capsys):
         cap = capsys.readouterr()
         assert 'Absolute path not found for executable: exedir' in cap.err
         assert 'Ensure $PATH environment variable contains common executable locations' in cap.err
+
+
+@pytest.fixture
+def stub_which(monkeypatch):
+    def apply(value='/bin/restorecon'):
+        monkeypatch.setattr(system, 'which', lambda x: value)
+    return apply
+
+
+class TestSetContext(object):
+
+    def setup(self):
+        try:
+            os.environ.pop('CEPH_VOLUME_SKIP_RESTORECON')
+        except KeyError:
+            pass
+
+    @pytest.mark.parametrize('value', ['1', 'True', 'true', 'TRUE', 'yes'])
+    def test_set_context_skips(self, stub_call, fake_run, value):
+        stub_call(('', '', 0))
+        os.environ['CEPH_VOLUME_SKIP_RESTORECON'] = value
+        system.set_context('/tmp/foo')
+        assert fake_run.calls == []
+
+    @pytest.mark.parametrize('value', ['0', 'False', 'false', 'FALSE', 'no'])
+    def test_set_context_doesnt_skip_with_env(self, stub_call, stub_which, fake_run, value):
+        stub_call(('', '', 0))
+        stub_which()
+        os.environ['CEPH_VOLUME_SKIP_RESTORECON'] = value
+        system.set_context('/tmp/foo')
+        assert len(fake_run.calls)
+
+    def test_set_context_skips_on_executable(self, stub_call, stub_which, fake_run):
+        stub_call(('', '', 0))
+        stub_which('restorecon')
```

Comment:
shouldn't this be `selinuxenabled`?

Reply:
No. In this test, the check for `selinuxenabled` passes correctly indicating that SELinux is enabled in the system, and the stub is for the `which()` function that will return `storecon` (without a leading `/`) indicating that the executable is not found, in which case the call is skipped.